pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/groupby/DictionaryBasedGroupKeyGenerator.java

```diff
@@ -101,7 +101,19 @@ public class DictionaryBasedGroupKeyGenerator implements GroupKeyGenerator {
 
   private final int _globalGroupIdUpperBound;
   private final RawKeyHolder _rawKeyHolder;
+  private final HolderType _usedHolderType;
```

Comment:
We can make `RawKeyHolder` extends `AutoCloseable`, and make this cleanup logic in `close()`. We don't need to maintain the `HolderType` here